### PR TITLE
CMake: Considerably improve summary message

### PR DESCRIPTION
**Pull request by tamiko**
This commit restructures the summary output at the end of the configure
phase to take latest results of UI and CLI development into account:

[1] Norman, D. A. (2002). "Emotion & Design: Attractive things work
    better". Interactions Magazine, ix (4). pp. 36–42. Retrieved 20 April 2014.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -163,5 +163,21 @@ ENDIF()
 #
 # And finally, print the configuration:
 #
-FILE(READ ${CMAKE_BINARY_DIR}/summary.log DEAL_II_LOG_SUMMARY)
-MESSAGE("${DEAL_II_LOG_SUMMARY}")
+FIND_PACKAGE(Perl QUIET)
+IF(PERL_FOUND AND DEAL_II_HAVE_BUNDLED_DIRECTORY)
+  EXECUTE_PROCESS(
+    COMMAND ls ${COWSAY_FOLDER}/cows
```
Comment (bangerth) [inline CMakeLists.txt]: Could you add some commentary about the meaning of the farm animal words?

Comment (bangerth) [inline CMakeLists.txt]: Realizing now what cowsay is, how about something like "Use the bundled version of the cowsay system to produce nicer output."

**Comment by bangerth** [comment]
OK, I was going to suggest that you first show some output because I really couldn't figure out whether you're just trying to be funny here or whether there's a real purpose to it. Then, upon some more consideration, the date dawned upon me. I'm half-tempted to merge it but that would just lead to more work. Why don't you show us how the output would have looked?

**Comment by guidokanschat** [comment]
Wolfgang just wrote what I wanted to ask you. I think showing the output would be great. And then, given the date and "cowsay"... maybe we better don't merge?

**Comment by bangerth** [comment]
I did review the whole patch, by the way, before I realized it all.

**Comment by tamiko** [comment]
```
 ______________________________________________________________________________________________ 
/ ###                                                                                          \
| #                                                                                            |
| #  deal.II configuration:                                                                    |
| #        CMAKE_BUILD_TYPE:       DebugRelease                                                |
| #        BUILD_SHARED_LIBS:      ON                                                          |
| #        CMAKE_INSTALL_PREFIX:   /usr/local                                                  |
| #        CMAKE_SOURCE_DIR:       /home/tamiko/workspace/dealii                               |
| #                                (version 8.3.pre, shortrev 91a7d5e)                         |
| #        CMAKE_BINARY_DIR:       /home/tamiko/workspace/dealii/build                         |
| #        CMAKE_CXX_COMPILER:     GNU 4.9.2 on platform Linux x86_64                          |
| #                                /usr/bin/c++                                                |
| #                                                                                            |
| #  Configured Features (DEAL_II_ALLOW_BUNDLED = ON, DEAL_II_ALLOW_AUTODETECTION = ON):       |
| #      ( DEAL_II_WITH_64BIT_INDICES = OFF )                                                  |
| #        DEAL_II_WITH_ARPACK set up with external dependencies                               |
| #        DEAL_II_WITH_BOOST set up with external dependencies                                |
| #        DEAL_II_WITH_BZIP2 set up with external dependencies                                |
| #        DEAL_II_WITH_CXX11 = ON                                                             |
| #      ( DEAL_II_WITH_HDF5 = OFF )                                                           |
| #        DEAL_II_WITH_LAPACK set up with external dependencies                               |
| #      ( DEAL_II_WITH_METIS = OFF )                                                          |
| #      ( DEAL_II_WITH_MPI = OFF )                                                            |
| #        DEAL_II_WITH_MUPARSER set up with external dependencies                             |
| #        DEAL_II_WITH_NETCDF set up with external dependencies                               |
| #        DEAL_II_WITH_OPENCASCADE set up with external dependencies                          |
| #      ( DEAL_II_WITH_P4EST = OFF )                                                          |
| #      ( DEAL_II_WITH_PETSC = OFF )                                                          |
| #      ( DEAL_II_WITH_SLEPC = OFF )                                                          |
| #        DEAL_II_WITH_THREADS set up with external dependencies                              |
| #      ( DEAL_II_WITH_TRILINOS = OFF )                                                       |
| #        DEAL_II_WITH_UMFPACK set up with external dependencies                              |
| #        DEAL_II_WITH_ZLIB set up with external dependencies                                 |
| #                                                                                            |
| #  Component configuration:                                                                  |
| #      ( DEAL_II_COMPONENT_DOCUMENTATION = OFF )                                             |
| #        DEAL_II_COMPONENT_EXAMPLES                                                          |
| #        DEAL_II_COMPONENT_MESH_CONVERTER                                                    |
| #      ( DEAL_II_COMPONENT_PACKAGE = OFF )                                                   |
| #      ( DEAL_II_COMPONENT_PARAMETER_GUI = OFF )                                             |
| #                                                                                            |
| #  Detailed information (compiler flags, feature configuration) can be found in detailed.log |
| #                                                                                            |
| #  Run  $ make info  to print a help message with a list of top level targets                |
| #                                                                                            |
\ ###                                                                                          /
 ---------------------------------------------------------------------------------------------- 
        \  ^___^
         \ (o o)\_______
           (___)\       )\/\
                ||----w |
                ||     ||
```

**Comment by bangerth** [comment]
Nicely done :-)

(You must know that I'm a big fan of cows. They roam all of the state I live in, and they TASTE delicious!)

**Comment by guidokanschat** [comment]
I'm really missing the "Like" button for Wolfgang's message

**Comment by bangerth** [comment]
You're welcome to express your feelings towards me or my message in the form you just chose ;-)

**Comment by tamiko** [comment]
@bangerth
> I did review the whole patch, by the way, before I realized it all.

Gotcha? :-]

**Comment by bangerth** [comment]
Well, almost :-)

**Pull request closed without merging**